File: app/src/main/java/com/rockon999/android/leanbacklauncher/wallpaper/FadeMaskView.java
package com.rockon999.android.leanbacklauncher.wallpaper;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapShader;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Shader.TileMode;
import android.util.AttributeSet;
import android.view.View;

public class FadeMaskView extends View {
    private Bitmap mMask;
    private Paint mPaint;

    public FadeMaskView(Context context) {
        this(context, null);
    }

    public FadeMaskView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public FadeMaskView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        this.mPaint = new Paint();
    }

    public void setBitmap(Bitmap maskImage) {
        this.mMask = maskImage;
        if (this.mMask != null) {
            this.mPaint.setShader(new BitmapShader(this.mMask, TileMode.REPEAT, TileMode.CLAMP));
        }
    }

    protected void onDraw(Canvas canvas) {
        if (this.mMask != null) {
            canvas.drawRect(0.0f, 0.0f, (float) getWidth(), (float) getHeight(), this.mPaint);
        }
    }
}

File: app/src/main/java/com/amazon/tv/leanbacklauncher/wallpaper/FadeMaskView.kt
package com.amazon.tv.leanbacklauncher.wallpaper

import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapShader
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.Shader.TileMode
import android.util.AttributeSet
import android.view.View

class FadeMaskView @JvmOverloads constructor(
    context: Context?,
    attrs: AttributeSet? = null,
    defStyle: Int = 0
) : View(context, attrs, defStyle) {
    private var mMask: Bitmap? = null
    private val mPaint: Paint = Paint()
    fun setBitmap(maskImage: Bitmap?) {
        mMask = maskImage
        if (mMask != null) {
            mPaint.shader = BitmapShader(mMask!!, TileMode.REPEAT, TileMode.CLAMP)
        }
    }

    override fun onDraw(canvas: Canvas) {
        if (mMask != null) {
            canvas.drawRect(0.0f, 0.0f, width.toFloat(), height.toFloat(), mPaint)
        }
    }

}